dbt/adapters/bigquery/impl.py

```diff
@@ -620,12 +620,13 @@ def _partitions_match(table, conf_partition: Optional[PartitionConfig]) -> bool:
             table_field = (
                 table.time_partitioning.field.lower() if table.time_partitioning.field else None
             )
+
             table_granularity = table.partitioning_type
             conf_table_field = conf_partition.field
             return (
-                table_field == conf_table_field
+                table_field.lower() == conf_table_field.lower()
```

Comment:
Are we also encountering the issue with the field names? Or is it just the granularity? I ask because I could see this creating an issue for users using case sensitive columns. It's a very small chance (e.g. `column_a` and `Column_A` on the same table), but if we can avoid it, even better.

Reply:
Good question!
Well the `lower()` on the right was also there before the regression so I'm putting it back.
I assume that if we lower right one, we should lower left one too to have a case insensitive match.

As far as I know (and retested), column names are case insensitive. So it makes sense to do a proper case insensitive match (and the test case I added is checking it).

Reply:
If the `.lower()` was there on one side, then I agree it should be there on the other. But it doesn't look like that's the case here: https://github.com/dbt-labs/dbt-bigquery/blob/eb58c7db58d389695f6f2dad8b4f4c8f5f4b5db3/dbt/adapters/bigquery/impl.py#L626. Am I looking at the wrong line?

You're correct that column names are case-insensitive. However, if you quote them then they become case sensitive. Again, it's a small chance, but that's what I'm guarding against.

Reply:
@mikealfare you're close, it's there: https://github.com/dbt-labs/dbt-bigquery/blob/eb58c7db58d389695f6f2dad8b4f4c8f5f4b5db3/dbt/adapters/bigquery/impl.py#L620-L621

So you're suggesting that there would be people using backticks in the dbt config headers?
Like this: 
```
{{
  config(
    partition_by={
      "field": "`updated_at`",
      "data_type": "date",
    }
  )
}}
```
I mean it could be... but then I think it's going to fail anyway as we wouldn't remove the backticks when doing the check anyway, right?

Reply:
Thanks for pointing me in the right direction. I'm comfortable with that then.

Reply:
Oh wait, we can't do `.lower()` here because `table_field` can be `None` from that line. But I'm comfortable adding it on the right side.

Reply:
Right, it's already lowered anyway, I'm removing it, good point!